packages/react-components/src/components/DevicesButton.tsx

```diff
@@ -371,15 +373,22 @@ export const generateDefaultDeviceMenuProps = (
 export const DevicesButton = (props: DevicesButtonProps): JSX.Element => {
   const { onRenderIcon } = props;
 
+  /* @conditional-compile-remove(rooms) */
+  const rolePermissions = _usePermissions();
+
   const localeStrings = useLocale().strings.devicesButton;
   const strings = { ...localeStrings, ...props.strings };
 
   const devicesButtonMenu =
     props.menuProps ?? generateDefaultDeviceMenuProps({ ...props, styles: props.styles?.menuStyles }, strings);
 
-  const onRenderOptionsIcon = (): JSX.Element => (
-    <HighContrastAwareIcon disabled={props.disabled} iconName="ControlButtonOptions" />
-  );
+  const onRenderOptionsIcon = (): JSX.Element => {
+    /* @conditional-compile-remove(rooms) */
+    if (!rolePermissions.microphoneButton && !rolePermissions.cameraButton) {
```

Comment:
In fact yeah option 2 won't work as designed because if they have no cameras registered then the icon may change when they wouldn't want it to, I think option 1 is best

Reply:
Done, moved permissions logic to composites